### PR TITLE
docs($provide): Decorator method in summary

### PR DESCRIPTION
**Pull request by ajaygm**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Docs update

**What is the current behavior? (You can also link to an open issue here)**
A short description of decorator method is missing.

**What is the new behavior (if this is a feature change)?**
A short description of decorator method added to the summary section.

**Does this PR introduce a breaking change?**
No

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Other information**:

Added short description for decorator method and link to it in the summary section.

### CODE REVIEW

File: src/auto/injector.js [hunk 1]
```diff
@@ -359,6 +359,8 @@ function annotate(fn, strictDi, name) {
  * * {@link auto.$provide#service service(class)} - registers a **constructor function**, `class`
  *     that will be wrapped in a **service provider** object, whose `$get` property will instantiate
  *      a new object using the given constructor function.
+ * * {@link auto.$provide#decorator decorator(name)} - registers a **service decorator** with the
```
Comment (Narretz) [inline src/auto/injector.js]: service decorator is not correct, as not only services can be decorated. You should call it **decorator function**. In the function signature, the decorator fn is also missing: decorator(name, decoratorFn)

**Comment by ajaygm** [comment]
Hi @Narretz 
Thanks for your review. I had picked the summary from the detailed description at the bottom of the code. Based on your comment, have tried to update the detailed description section as well. Please let me know if it looks ok now.
Thanks!

File: src/auto/injector.js [hunk 2]
```diff
@@ -359,6 +359,8 @@ function annotate(fn, strictDi, name) {
  * * {@link auto.$provide#service service(class)} - registers a **constructor function**, `class`
  *     that will be wrapped in a **service provider** object, whose `$get` property will instantiate
  *      a new object using the given constructor function.
+ * * {@link auto.$provide#decorator decorator(name, decorFn)} - registers a **decorator function**,
+ *      `decorFn`, with the {@link auto.$injector $injector}, `name` the name of the provider to decorate.
```
Comment (Narretz) [inline src/auto/injector.js]: , `name` the name of the provider to decorate. -> ; `name` is the name of the provider to decorate.

File: src/auto/injector.js [hunk 3]
```diff
@@ -615,18 +617,18 @@ function annotate(fn, strictDi, name) {
  * @name $provide#decorator
  * @description
  *
- * Register a **service decorator** with the {@link auto.$injector $injector}. A service decorator
- * intercepts the creation of a service, allowing it to override or modify the behavior of the
- * service. The object returned by the decorator may be the original service, or a new service
- * object which replaces or wraps and delegates to the original service.
+ * Register a **decorator function** with the {@link auto.$injector $injector}. A decorator function
+ * intercepts the creation of a provider, allowing it to override or modify the behavior of the
```
Comment (Narretz) [inline src/auto/injector.js]: provider, -> provider (and also directives and filters),

**Comment by Narretz** [comment]
@ajaygm Thanks for the update. I left a few notes.

Can you also add a ink to the decorator guide in the detailed description before the example?

**Pull request closed without merging**